es-ds-components/components/es-toggle.vue

```diff
@@ -0,0 +1,170 @@
+<script setup lang="ts">
+import { SwitchRoot, SwitchThumb } from 'reka-ui';
+
+const model = defineModel<boolean>({
+    default: false,
+});
+
+defineProps({
+    disabled: {
+        type: Boolean,
+        default: false,
+    },
+    ariaLabel: {
+        type: String,
+        default: 'es-toggle',
+        required: false,
+    },
+    hasValue: {
+        type: Boolean,
+        default: false,
+    },
+    onValue: {
+        type: String,
+        default: 'ON',
+        required: false,
+    },
+    offValue: {
+        type: String,
+        default: 'OFF',
+        required: false,
+    },
+});
+
+// Workaround: Reka UI components don't play nice with the Nuxt 3.16+ dev server side rendering while using NPM link
+const wrappingComponent = import.meta.dev ? resolveComponent('ClientOnly') : 'span';
+</script>
+
+<template>
+    <component :is="wrappingComponent">
+        <div
+            class="es-toggle-container"
+            :class="{ 'has-value': hasValue, 'has-text': $slots.text }">
+            <div class="es-toggle-wrapper">
+                <switch-root
+                    v-model="model"
+                    :disabled="disabled"
+                    :aria-label="ariaLabel"
+                    :aria-labelledby="hasValue ? 'es-toggle-value' : undefined"
+                    class="es-toggle">
+                    <switch-thumb class="es-toggle-thumb" />
+                </switch-root>
+                <p
+                    v-if="hasValue"
+                    id="es-toggle-value"
+                    class="es-toggle-value">
+                    {{ model ? onValue : offValue }}
+                </p>
+            </div>
+            <div v-if="$slots.text">
+                <slot name="text" />
+            </div>
```

Comment:
If you check out the current switch usage in es-onboarding, you'll note that the "description" is linked to the toggle in such a way that when I click on the text, it switches the toggle on and off. I think that for accessibility and usability purposes it would make sense to have that same behavior here.

https://github.com/user-attachments/assets/0b99ac57-0b6a-4e08-9c65-511c276c7cfc